webapp/src/app/ui/avatar/avatar.component.ts

```diff
@@ -1,51 +1,31 @@
-import { Component, computed, input, signal } from '@angular/core';
+import { Component, computed, inject, InjectionToken, input, signal } from '@angular/core';
 import type { ClassValue } from 'clsx';
-import type { VariantProps } from 'class-variance-authority';
 import { cn } from 'app/utils';
-import { cva } from 'app/storybook.helper';
-import { NgOptimizedImage } from '@angular/common';
 
-const [avatarVariants, args, argTypes] = cva('relative flex shrink-0 overflow-hidden rounded-full', {
-  variants: {
-    size: {
-      default: 'h-10 w-10',
-      sm: 'h-6 w-6 text-xs',
-      lg: 'h-14 w-14 text-lg'
-    }
-  },
-  defaultVariants: {
-    size: 'default'
-  }
-});
+const AvatarToken = new InjectionToken<AvatarComponent>('AvatarToken');
 
-export { args, argTypes };
+export function injectAvatar(): AvatarComponent {
+  return inject(AvatarToken);
+}
 
-interface AvatarVariants extends VariantProps<typeof avatarVariants> {}
+type ImageLoadingStatus = 'idle' | 'loading' | 'loaded' | 'error';
 
 @Component({
   selector: 'app-avatar',
   standalone: true,
-  imports: [NgOptimizedImage],
-  templateUrl: './avatar.component.html'
+  template: `<ng-content />`,
+  providers: [{ provide: AvatarToken, useExisting: AvatarComponent }],
+  host: {
+    '[class]': 'computedClass()'
+  }
 })
-export class AppAvatarComponent {
-  class = input<ClassValue>('');
-  size = input<AvatarVariants['size']>('default');
-
-  src = input.required<string>();
-  alt = input<string>('');
-  imageClass = input<string>('');
-  fallback = input<string>('https://placehold.co/56');
+export class AvatarComponent {
```

Comment:
Nit: for consistency reasons this should probably use the `App`-prefix like our other component classes, but if I learned Angular correctly only the component-selector really matters anyways.

Reply:
Yeah, only the component selector is important. I think we could just omit `App` everywhere, when I used `ng g c avatar` it did not come with the prefix for the class. For consistency we'd need to adjust the other components, what do you think?

Reply:
Agreed, let's remove it for all classes.